controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +199,105 @@ func (r *PolicyAutomationReconciler) Reconcile(
 			} else {
 				log.Info("All clusters are compliant. Doing nothing.")
 			}
+		} else if policyAutomation.Spec.Mode == "everyEvent" {
+			log := log.WithValues("mode", "everyEvent")
+			targetList := common.FindNonCompliantClustersForPolicy(policy)
+			// original eventTime is earlier than automationStartTime
+			eventTime := time.Now().UTC()
+			eventTimeStr := eventTime.Format(time.RFC3339)
+			if len(targetList) > 0 {
+				// delayAfterRunSeconds default value is zero
+				delayAfterRunSeconds := policyAutomation.Spec.DelayAfterRunSeconds
+
+				eventMap := policyAutomation.Status.ClustersWithEvent
+				// final target clusters list that the new ansible job will be created on
+				trimmedTargetList := []string{}
+
+				if len(eventMap) > 0 { // check existing automation status on each cluster
+					// convert slice targetList to map for search efficiency
+					targetListMap := map[string]bool{}
+					for _, target := range targetList {
+						targetListMap[target] = true
+					}
+
+					for clusterName := range eventMap {
+						clusterEvent := eventMap[clusterName]
+						oldAutomationStartTime, err := time.Parse(time.RFC3339, clusterEvent.AutomationStartTime)
+						if err != nil {
+							log.Error(err, "Failed to convert AutomationStartTime")
+
+							return reconcile.Result{}, err
```

Comment:
This doesn't seem like an error that we can recover from. I imagine this would only happen if a user modified the status manually. In this case, I would delete the entry from `eventMap` and continue.